web/react/components/posts_view.jsx

```diff
@@ -0,0 +1,301 @@
+// Copyright (c) 2015 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+const UserStore = require('../stores/user_store.jsx');
+const Utils = require('../utils/utils.jsx');
+const Post = require('./post.jsx');
+
+export default class PostsView extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.handleScroll = this.handleScroll.bind(this);
+        this.isAtBottom = this.isAtBottom.bind(this);
+        this.loadMorePostsTop = this.loadMorePostsTop.bind(this);
+        this.postsRerendered = this.postsRerendered.bind(this);
+        this.createPosts = this.createPosts.bind(this);
+        this.updateScrolling = this.updateScrolling.bind(this);
+        this.handleResize = this.handleResize.bind(this);
+
+        this.jumpToPostNode = null;
+        this.wasAtBottom = true;
+        this.scrollHeight = 0;
+    }
+    static get SCROLL_TYPE_FREE() {
+        return 1;
+    }
+    static get SCROLL_TYPE_BOTTOM() {
+        return 2;
+    }
+    static get SIDEBAR_OPEN() {
+        return 3;
+    }
+    isAtBottom() {
+        return ((this.refs.postlist.scrollHeight - this.refs.postlist.scrollTop) === this.refs.postlist.clientHeight);
+    }
+    handleScroll() {
+        // HACK FOR RHS -- REMOVE WHEN RHS DIES
+        const childNodes = this.refs.postlistcontent.childNodes;
+        for (let i = 0; i < childNodes.length; i++) {
+            // If the node is 1/3 down the page
+            if (childNodes[i].offsetTop > (this.refs.postlist.scrollTop + (this.refs.postlist.offsetHeight / 3))) {
+                this.jumpToPostNode = childNodes[i];
+                break;
+            }
+        }
+        this.wasAtBottom = this.isAtBottom();
+
+        // --- --------
+
+        this.props.postViewScrolled(this.isAtBottom());
+        this.prevScrollHeight = this.refs.postlist.scrollHeight;
+    }
+    loadMorePostsTop() {
+        this.props.loadMorePostsTopClicked();
+    }
+    postsRerendered() {
+    }
```

Comment:
I left this in because I didn't wan't to touch the post stuff too much when I was changing everything. But since this function turned out to be completely pointless I will remove.